src/main/java/me/snowdrop/fallback/AnnotationAwareFallbackOperationsInterceptor.java

```diff
@@ -119,42 +102,53 @@ private Method findTargetMethod(Class fallbackClass, String fallbackMethod) {
         return ReflectionUtils.findMethod(fallbackClass, fallbackMethod, ExecutionContext.class);
     }
 
-    private MethodInterceptor getDelegate(Object target, Fallback fallback) {
-        final String methodName = getMethodName(fallback);
-
-        if (fallback.value().equals(void.class)) { //we need to use the fallback method of the target class
-            final Method targetFallbackMethod = findTargetMethod(getTargetClass(target), methodName);
-
-            if (targetFallbackMethod == null) {
-                throw new IllegalArgumentException(target + " does not contain a method named '" + methodName + "'");
-            }
-
-            return new NonStaticErrorHandlerFallbackOperationsInterceptor(targetFallbackMethod, target);
-        }
-        else {
-            final Method targetFallbackMethod = findTargetMethod(fallback.value(), methodName);
-
-            if (targetFallbackMethod == null) {
-                throw new IllegalArgumentException(fallback.value() + " does not contain a static method named '" + methodName + "'");
-            }
-
-            if (Modifier.isStatic(targetFallbackMethod.getModifiers())) {  //in this a static method is used
-                return new StaticErrorHandlerFallbackOperationsInterceptor(targetFallbackMethod);
-            }
-            else { //finally we attempt to find a Spring bean using the class supplied
-
-                try {
-                    return new NonStaticErrorHandlerFallbackOperationsInterceptor(
-                            targetFallbackMethod,
-                            beanFactory.getBean(fallback.value())
-                    );
-                } catch (BeansException e) {
-                    throw new UnsupportedOperationException("Unable to retrieve bean of class " + fallback.value(), e);
-                }
-
-
-            }
-        }
+    private MethodInterceptor doGetMethodInterceptor(Object target, Set<Fallback> fallbacks) {
+
+
+        final List<FallbackInterceptor.Configuration> configurationList =
+                fallbacks
+                        .stream()
+                        .map(f -> {
+                            final String methodName = getMethodName(f);
+                            if (f.value().equals(void.class)) { //we need to use the fallback method of the target class
```

Comment:
Can you use `void.class.equals(f.value())` to avoid NPE in case of f.value() returns a null.